NewForce-Cohort-8/tabloidfullstack-suplex-syndicate
# Lv reactivate user

### Description
Added ability for admins to deactivate users and prevent those users from logging in
[Trello Ticket 26](https://trello.com/c/ogN1J6nA/26-26-reactivate-a-user-profile)


### Testing Instructions
1. `git add` and `git commit` whatever you're working on in your branch
2. `git checkout main`
3. `git fetch -a`
4. `git checkout lv-reactivate-user`
5. If you don't see the most recent version of my files `git pull origin lv-reactivate-user`
6. If you have not created the database please do so and follow Steps 7-8 otherwise proceed to Step 9
7. Run this [SQL script](https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/01_Tabloid_Create_DB.sql) to create the initial database
8. Run this [SQL script](https://github.com/NewForce-Cohort-8/tabloidfullstack-suplex-syndicate/blob/main/SQL/02_Tabloid_Seed_Data.sql) to seed the initial data
9. If you tested my previous pull request run the following SQL query `﻿UPDATE UserProfile SET UserStatusId = 1;` and proceed to step 13
10. Run these commands in a new SQL query, executing each command one by one
`﻿ALTER TABLE UserProfile ADD UserStatusId int;`
`CREATE TABLE UserStatus (
	Id int,
	Name varChar(50)
);`
`﻿INSERT INTO UserStatus (Id, Name) VALUES (1, 'Active');
INSERT INTO UserStatus(Id, Name) VALUES (2, 'Deactivated');`
`﻿UPDATE UserProfile SET UserStatusId = 1;`
11. Afterwards right click on the UserProfile table in the database and select **View Data** and confirm that a UserStatusId Column has been added and all users have a UserStatusId of 1
12. Right Click on the UserStatus table and select **View Data** to confirm that the data table has been created and seeded with the above information.
13. Open the TabloidFullStack.sln with Visual Studio
14. At the top of Visual Studio, select https as the launcher and click the green play button
15. The Swagger UI should appear in the browser
16. Confirm there is a PUT, a POST, and 4 GET calls under the UserProfile controller
17. If the React app isn't already running, in your terminal `npm start` otherwise refresh the React App
18. Login with the email `foo@bar.com`
19. Select `User Profiles`
20. On Aharon Grzeskowski's row, select the red `Deactivate` button
21. Note that Aharon has a user type of Admin
22. Confirm you have been redirected to **http://localhost:3000/UserProfiles/5/Deactivate**
23. Select the red `Deactivate` button
24. You should be redirected back to the User Profile list
25. Confirm that Aharon Grzeskowski no longer has a red Deactivate button in his row and instead has a green Reactivate button
26. Select **View Deactived** 
27. Aharon should be the only profile listed
28. Select **View Active**
29. Note that Aharon is not listed
30. Log out
31. try to log in with the email `agrzeskowski3@fc2.comx`
32. You should recieve an invalid email or password warning and still be on the login page
33. Log in with the email `foo@bar.com`
34. Select User Profiles
35. Select View Deactivated
36. Select the green Reactivate button in Aharon's row
37. Confirm that you have been redirected to `http://localhost:3000/UserProfiles/5/Reactivate`
38. Select the green Reactivate button
39. you should be redirected back to the user profile list
40. confirm that Aharon has a red Deactivate button again
41. Confirm that Aharon still has a user type of Admin
42. Log out
43. Log in with the email `agrzeskowski3@fc2.comx`
44. You should be able to use the app as any admin should.

### Checklist
